src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -270,274 +263,25 @@ private Task<bool> ReadAsynchronously()
             return TaskUtils.GetTaskForSynchronousOperation<bool>(this.ReadImplementation);
         }
 #endif
-
-        /// <summary>
-        /// Continues reading from the batch message payload.
-        /// </summary>
-        /// <returns>true if more items were read; otherwise false.</returns>
-        private bool ReadImplementation()

```

Comment:
>ReadImplementation [](start = 21, length = 18)

The *Implementation methods should be private methods in the base class that deal with common logic (primarily statemanagement/validation logic).  Those methods should call into abstract methods that are overridden in the derived classes to do the actual reading.